homeassistant/components/honeywell/climate.py

```diff
@@ -482,31 +482,33 @@ async def async_turn_aux_heat_off(self) -> None:
 
     async def async_update(self) -> None:
         """Get the latest state from the service."""
-        try:
-            await self._device.refresh()
-            self._attr_available = True
-            self._retry = 0
 
-        except UnauthorizedError:
+        async def _async_update(login=False) -> None:
+            """Get the latest state from the service."""
             try:
-                await self._data.client.login()
                 await self._device.refresh()
                 self._attr_available = True
                 self._retry = 0
 
-            except (
-                SomeComfortError,
-                ClientConnectionError,
-                asyncio.TimeoutError,
-            ):
+            except UnauthorizedError:
+                if not login:
+                    try:
+                        await self._data.client.login()
```

Comment:
Instead of all this nesting can't we put the login with it's retries in it's own function. Looks cleaner

Reply:
I thought we were trying to reduce duplicate code.  If I put the login in it's own function it's just the same duplicate code as before :( 
Lemme see what it will look like as a function - Probably share first thing tomorrow as my dev docker has crashed again and I need to rebuild it.

Reply:
Refactored to a _login() function: Is this what you had in mind?